src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,173 @@
+                leftOperand = NullableAlwaysHasValue(leftOperand) ?? leftOperand;
+
+                if (node.RightOperand is null)
+                {
+                    if (leftOperand.Type.IsNullableType())
+                    {
+                        // Case 4:              "nullableLeft.."
+                        // translates to:       nullableLeft.HasValue ? new Nullable(Range.FromStart(nullableLeft.GetValueOrDefault())) : default
+
+                        MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__FromStart);
+                        return LiftRangeExpression(node.Syntax, node.Type, method, leftOperand);
+                    }
+                    else
+                    {
+                        // Case 5:              "left.."
+                        // translates to:       Range.FromStart(left)
+
+                        MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__FromStart);
+                        ImmutableArray<BoundExpression> arguments = ImmutableArray.Create(leftOperand);
+                        return MakeCall(node.Syntax, rewrittenReceiver: null, method, arguments, node.Type);
+                    }
+                }
+
+                BoundExpression rightOperand = VisitExpression(node.RightOperand);
+                rightOperand = NullableAlwaysHasValue(rightOperand) ?? rightOperand;
+
+                if (!leftOperand.Type.IsNullableType() && !rightOperand.Type.IsNullableType())
+                {
+                    // Case 6:              "left..right"
+                    // translates to:       Range.Create(left, right)
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__Create);
+                    ImmutableArray<BoundExpression> arguments = ImmutableArray.Create(leftOperand, rightOperand);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, arguments, node.Type);
+                }
+                else
+                {
+                    // Case 7:              "nullableLeft..nullableRight"
+                    // translates to:       nullableLeft.HasValue && nullableRight.HasValue ? new Nullable(Range.Create(nullableLeft.GetValueOrDefault(), nullableRight.GetValueOrDefault())) : default
+
+                    // Case 8:              "nullableLeft..right"
+                    // translates to:       nullableLeft.HasValue ? new Nullable(Range.Create(nullableLeft.GetValueOrDefault(), right)) : default
+
+                    // Case 9:              "left..nullableRight"
+                    // translates to:       nullableRight.HasValue ? new Nullable(Range.Create(left, nullableRight.GetValueOrDefault())) : default
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__Create);
```

Comment:
>MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__Create); [](start = 20, length = 110)

It feels like this can be moved out of the ```if``` and ```else```.